doc/macsec/MACsec_hld.md

```diff
@@ -762,6 +781,128 @@ All boxes with black edge are components of virtual SAI and all boxes with purpl
+attr.value.aclfield.enable = true;
+```
+
+- Bypass frames
+
+``` c++
+attr.id = SAI_ACL_ENTRY_ATTR_ACTION_PACKET_ACTION;
+attr.value.aclaction.parameter.s32 = SAI_PACKET_ACTION_FORWARD;
+attr.value.aclaction.enable = true;
+```
+
+- Drop frames
+
+``` c++
+attr.id = SAI_ACL_ENTRY_ATTR_ACTION_PACKET_ACTION;
+attr.value.aclaction.parameter.s32 = SAI_PACKET_ACTION_DROP;
+attr.value.aclaction.enable = true;
+```
+
+- **React encrypted ingress PFC frame**: The default behavior of MACsec engine should decrypt encrypted frames and forward it to the upper layer. So, there is no extra configuration to encrypted ingress PFC frame.
+- **React clear ingress PFC frame**: We don't want to assume an implicit default behavior to clear PFC frames in different MACsec engines. So, the ACL entry should has explicit configuration of **identify PFC frames** and **bypass frames** if we want to accept clear ingress PFC frames.
+- **Deny clear ingress PFC**: Due to the same reason with **Send clear ingress PFC frame**, the ACl entry should has explicit configuration of **identify PFC frames** and **Drop frames** if we want to deny clear ingress PFC frames.
+- **Send encrypted egress PFC frame**: The default behavior of MACsec engine should encrypt all frames and forward it to the peer side. So, there is no extra configuration to egress PFC frame.
+- **Send clear egress PFC frame**: If the switch want to send clear egress PFC frames, the PFC frames should bypass the MACsec engine. The ACL entry should **identify PFC Frames** first and **bypass frames**.
+
+##### 3.5.1.2 PFC mode
+
+- **Bypass mode**: The switch should react clear and encrypted ingress PFC frames and should send clear egress PFC frames.
+
+![pfc_bypass_mode](images/pfc_bypass_mode.png)  
+
+- **Encrypt mode**: The switch should react clear and encrypted PFC frames, send encrypted PFC frames.
+
+![pfc_encrypt_mode](images/pfc_encrypt_mode.png)  
+
+- **Strict Encrypt mode**: The switch should only react encrypted PFC frames, send encrypted PFC frames.
+
+![pfc_strict_encrypt_mode](images/pfc_strict_mode.png)  
+
+***Why encrypt mode? If the peer switch can only send clear PFC but react both, Encrypt mode is safer than Bypass mode.***
+
+##### 3.5.1.3 PFC counter
+
+In the strict encrypt mode, to count the dropped ingress clear PFC frames per port and the counter will be added into the PFC ACL entry mentioned above.
+
+``` c++
+sai_attribute_t attr;
+vector<sai_attribute_t> counter_attrs;
+
+attr.id = SAI_ACL_COUNTER_ATTR_TABLE_ID;
+attr.value.oid = acl_table_oid;
+counter_attrs.push_back(attr);
+
+attr.id = SAI_ACL_COUNTER_ATTR_ENABLE_PACKET_COUNT;
+attr.value.booldata = true;
+counter_attrs.push_back(attr);
+
+// If in Gearbox mode, the switch_id is the gearbox id, otherwise it's the asic switch id
+sai_acl_api->create_acl_counter(&counter_id, switch_id, (uint32_t)counter_attrs.size(), counter_attrs.data());
+```
+
+***Per queue counter of PFC needs to be discussed with vendors further.***
```

Comment:
if the current cli ("show pfc counters") can be used, please state that. Also mention that there is currently no way to get separate counts for clear vs encrypted packets received in 'bypass' and 'encrypt' mode